scripts/bash/gitlab_helpers.sh

```diff
@@ -0,0 +1,393 @@
+    gitlab-rails runner "Ci::InstanceVariable.where(key: 'CONTAINER_REGISTRY_NAMESPACE').update(value: '$GCP_PROJECT_ID').save"
+    gitlab-rails runner "Ci::InstanceVariable.where(key: 'CONTAINER_REGISTRY_HOST').update(value: 'gcr.io').save"
+}
+
+
+seed_instance_reg_token () {
+    local logName=$1
+    # Seed shared runners registartion token
+    GITLAB_RUNNER_REG_SECRET=`curl -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/instance/attributes/gitlab-ci-runner-registration-token-secret`
+    logger $logName "INFO" "Reading shared runners registration token from secret $GITLAB_RUNNER_REG_SECRET"    
+    GITLAB_RUNNER_REG=`gcloud secrets versions access latest --secret=$GITLAB_RUNNER_REG_SECRET`
+    logger $logName "INFO" "Seeding shared runners registration token"
+    errMsg=$(gitlab-rails runner "appset = Gitlab::CurrentSettings.current_application_settings; appset.set_runners_registration_token('$GITLAB_RUNNER_REG'); appset.save!" 2>&1)
+    get_last_error $logName $? $ERR_ACTION_CONT "$errMsg" 
+}
+
+
+seed_gitlab_root_pwd () {
+    local logName=$1
+    # Seed gitlab root password
+    GITLAB_INITIAL_ROOT_PASSWORD_SECRET=`curl -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/instance/attributes/gitlab-initial-root-pwd-secret`
+    logger $logName "INFO" "Reading gitlab root password from secret $GITLAB_INITIAL_ROOT_PASSWORD_SECRET"
+    GITLAB_INITIAL_ROOT_PASSWORD=`gcloud secrets versions access latest --secret=$GITLAB_INITIAL_ROOT_PASSWORD_SECRET`
+    logger $logName "INFO" "Seeding gitlab root password"
+    errMsg=$(gitlab-rails runner "user = User.find_by_username('root'); user.password = '$GITLAB_INITIAL_ROOT_PASSWORD'; user.password_confirmation = '$GITLAB_INITIAL_ROOT_PASSWORD'; user.save!" 2>&1)
+    get_last_error $logName $? $ERR_ACTION_CONT "$errMsg"
+}
+
+
+
+create_groups () {
+    local logName=$1
+    # Create repo groups (ci, community, private)
+    logger $logName "INFO" "Creating groups: ci, community, private"
+    errMsg=$(gitlab-rails runner "Groups::CreateService.new(User.find_by_id(1), params = {name: 'CI CD Tools', path: 'ci', visibility_level: 10}).execute" 2>&1)
+    get_last_error $logName $? $ERR_ACTION_CONT "$errMsg"
+    errMsg=$(gitlab-rails runner "Groups::CreateService.new(User.find_by_id(1), params = {name: 'Community Tools', path: 'community', visibility_level: 10}).execute" 2>&1)
+    get_last_error $logName $? $ERR_ACTION_CONT "$errMsg"
+    errMsg=$(gitlab-rails runner "Groups::CreateService.new(User.find_by_id(1), params = {name: 'Private Tools', path: 'private', visibility_level: 10}).execute" 2>&1)
+    get_last_error $logName $? $ERR_ACTION_CONT "$errMsg"
+}
+
+
+import_scallopsRecipes () {
+    local logName=$1
+    # Import SCALLOPS-RECIPES project repo
+    SCALLOPS_RECIPES_GIT_URL=`curl -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/instance/attributes/scallops-recipes-git-url`
```

Comment:
use readonly definition

Reply:
What do you mean?

Reply:
Is this variable should change? if not make it a readonly variable.

Reply:
Changed